src/goto-programs/goto_function.h

```diff
@@ -118,6 +119,13 @@ class goto_functiont
   void validate(const namespacet &ns, const validation_modet vm) const
   {
     body.validate(ns, vm);
+    source_locationt nil_location;
+    nil_location.make_nil();
+    forall_subtypes(it, type)
+    {
+      find_typetag_in_namespace(*it, nil_location, ns, vm);
+    }
```

Comment:
Hmm, I don't get the impression that this is doing much of a "find." Couldn't you actually use one of the `find_*` functions to find the identifiers, and then check all of those that you have found?

Reply:
@tautschnig Ok, but this last bit required a bit more code change so I pushed it as a new commit and will squash once approved.